### PR TITLE
feat: clean template doc and template's project README

### PR DESCRIPTION
**Pull request by esciara**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,20 +1,477 @@
...
+indent_style = space
+insert_final_newline = false
+max_line_length = 88
+tab_width = 4
+ij_continuation_indent_size = 8
+ij_formatter_off_tag = @formatter:off
+ij_formatter_on_tag = @formatter:on
+ij_formatter_tags_enabled = false
+ij_smart_tabs = false
+ij_wrap_on_typing = false
```
Comment (slequeux) [inline .editorconfig]: Properties that are specific to Idea seems to be the opposite of what .editorconfig should do.
There should not be properties that are IDE specific.

File: Makefile [hunk 2]
```diff
@@ -66,6 +66,10 @@ clean:
 # generate project
 #################################################################
 
+overwrite:
+	rm -rf $(COOKIECUTTER_OUTPUT_DIR)/behave* $(COOKIECUTTER_OUTPUT_DIR)/features
```
Comment (slequeux) [inline Makefile:71]: Add a comment to explain that this is a bug workaround

File: Makefile [hunk 3]
```diff
@@ -66,6 +66,10 @@ clean:
...
+	poetry run cookiecutter --overwrite-if-exists --default-config --no-input -o $(BUILD_DIR) .
+
 refresh: clean
 	poetry run cookiecutter --overwrite-if-exists --default-config --no-input -o $(BUILD_DIR) .
```
Comment (slequeux) [inline Makefile]: Remove `--overwrite-if-exists` to explain why there are 2 targets (refresh and overwrite)

File: README.rst [hunk 4]
```diff
@@ -2,51 +2,122 @@
...
+.. image:: https://github.com/opinionated-digital-center/python-library-project-generator/workflows/Test%20and%20make%20release/badge.svg
+    :target: https://github.com/opinionated-digital-center/python-library-project-generator/actions?query=workflow%3A%22Test+and+make+release%22
+
+.. image:: https://github.com/opinionated-digital-center/python-library-project-generator/workflows/Test%20GitLab%20Pipeline/badge.svg
+    :target: https://github.com/opinionated-digital-center/python-library-project-generator/actions?query=workflow%3A%22Test+GitLab+Pipeline%22
+
+* GitHub repo: https://github.com/opinionated-digital-center/python-library-project-generator .
+* Free software: MIT license.
+
+The "Opinionated Python Library Project Template" is a fully tested, feature rich
```
Comment (slequeux) [inline README.rst:14]: Use backquotes instead of double quotes ?

File: README.rst [hunk 5]
```diff
@@ -2,51 +2,122 @@
...
+`Cookiecutter`_ scaffold to kick start your Python library and command line interface
+development.
+
+It is "opinionated" as it reflects strong structural and tooling choices,
+which we believe help developers being more effective within Digital Centers. It can
+however seamlessly be used for Open Source development (works also out-of-the-box with
+gitlab.com, github.com, pypi.org and readthedocs.org, as a standalone project).
```
Comment (slequeux) [inline README.rst]: Add links for gitlab, github, pypi and readthedocs ?

File: README.rst [hunk 6]
```diff
@@ -2,51 +2,122 @@
...
+
+In addition to the intrinsic features offered by the template (see "Features list"
```
Comment (slequeux) [inline README.rst]: Use anchor and link for `features list` ?

File: README.rst [hunk 7]
```diff
@@ -2,51 +2,122 @@
...
+below), the specific features that make it suitable and effective for Digital Centers
+are the abilities to:
+
+* Use self-hosted GitLab and GitHub.
+* Publish to self-hosted package repositories.
+* Reuse centralised and optimised pipeline elements. (available soon)
+* Reuse other centralised "toolbox" items.
```
Comment (slequeux) [inline README.rst]: Maybe be more specifi for these items

File: README.rst [hunk 8]
```diff
@@ -2,51 +2,122 @@
...
+
+The argumentation behind the Opinionated Digital Center's choices (for this and other
+projects) are/will be gradually documented as Architecture Decision Records, either
+on `ODC's ADR repository <https://github.com/opinionated-digital-center/architecture-decision-record>`_
+for transversal decisions, or on each individual repository's ``docs/adr`` directory
+(`here <https://github.com/opinionated-digital-center/python-library-project-generator/tree/master/docs/adr>`_
```
Comment (slequeux) [inline README.rst]: Is it possible to use relative links to point to the ADRs directory of this branch ?

File: README.rst [hunk 9]
```diff
@@ -2,51 +2,122 @@
...
+* The `Test GitLab Pipeline <https://github.com/opinionated-digital-center/python-library-project-generator/actions?query=workflow%3A%22Test+GitLab+Pipeline%22>`_
+  pipeline (GitHub available soon) to:
+
+  * Generate a project with the default options.
+  * Test that the generated project's GitLab pipeline runs successfully, including:
+
+    * Run the project's built-in tests and checks on the project's pipeline.
+    * Automatically bump the package's version following `Semantic Versioning`_ (based
+      on commit messages following the `Angular Commit Message Guideline`_, a subset
+      of the `Conventional Commits specification`_).
```
Comment (slequeux) [inline README.rst]: Maybe we could simplify this by pointing to the corresponding ADR

File: {{cookiecutter.project_slug}}/.editorconfig [hunk 10]
```diff
@@ -1,20 +1,477 @@
...
+indent_style = space
+insert_final_newline = false
+max_line_length = 88
+tab_width = 4
+ij_continuation_indent_size = 8
+ij_formatter_off_tag = @formatter:off
+ij_formatter_on_tag = @formatter:on
+ij_formatter_tags_enabled = false
+ij_smart_tabs = false
+ij_wrap_on_typing = false
```
Comment (slequeux) [inline {{cookiecutter.project_slug}}/.editorconfig]: Same remark as the one for `.editorconfig` in root dir of the repo

File: .editorconfig (hunk 1 above)
Comment (esciara) [reply .editorconfig]: done

File: Makefile (hunk 2 above)
Comment (esciara) [reply Makefile:71]: done

File: Makefile (hunk 3 above)
Comment (esciara) [reply Makefile]: done

File: README.rst (hunk 5 above)
Comment (esciara) [reply README.rst]: done

File: README.rst (hunk 7 above)
Comment (esciara) [reply README.rst]: done

File: README.rst (hunk 4 above)
Comment (esciara) [reply README.rst:14]: This is done on purpose. It is not a library name, a cli, a raw tool name, but the project title, hence the "".

File: README.rst (hunk 6 above)
Comment (esciara) [reply README.rst]: ~I can't make it work on github... `rst` is not as well supported as `md`... we might want to switch in the future for documents on the `root` directory.~ 

Found the solution: done.

File: README.rst (hunk 8 above)
Comment (esciara) [reply README.rst]: done

File: README.rst (hunk 9 above)
Comment (esciara) [reply README.rst]: done

File: {{cookiecutter.project_slug}}/.editorconfig (hunk 10 above)
Comment (esciara) [reply {{cookiecutter.project_slug}}/.editorconfig]: done

**Pull request merged**